scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -797,72 +885,102 @@ private ClickHandler clickHandlerForCheckButton() {
-                }
-                if (startAtRadioButton.getValue()) {
-                    DateTimeFormat dateTimeFormat = DateTimeFormat.getFormat(ISO_8601_FORMAT);
-                    String iso8601DateStr = dateTimeFormat.format(dateChooser.getData());
-                    startAtParameter.setValue(iso8601DateStr);
-                    hiddenPane.add(startAtParameter);
-                }
+                setAllValuesAsString();
+                handleStartAt();
                 enableValidation(true);
+                handleCompletedSubmission();
+                variablesActualForm.submit();
+                displayLoadingMessage();
+
+            }
+
+            private void handleCompletedSubmission() {
                 variablesActualForm.addSubmitCompleteHandler(new SubmitCompleteHandler() {
                     @Override
                     public void onSubmitComplete(SubmitCompleteEvent event) {
                         String result = event.getResults();
-                        if (result == null) {
-                            GWT.log("Unexpected empty result");
-                            displayErrorMessage("Unexpected empty result");
-                            return;
-                        }
-
-                        if (result.startsWith(SubmitEditServlet.ERROR)) {
-                            displayErrorMessage(result.substring(SubmitEditServlet.ERROR.length()));
+                        if (isResultNull(result) || ifResultIsAnError(result)) {
                             return;
                         }
 
                         try {
                             JSONValue json = JSONParser.parseStrict(result);
                             JSONObject obj = json.isObject();
-                            if (obj != null) {
-                                if (obj.containsKey("valid")) {
-                                    if (((JSONBoolean) obj.get("valid")).booleanValue()) {
-                                        if (obj.containsKey("updatedVariables")) {
-                                            updateVariables(obj.get("updatedVariables"));
-                                            redrawVariables();
-                                        }
-                                        GWT.log("Job validated");
-                                        displayInfoMessage("Job is valid");
-                                    } else if (obj.containsKey("errorMessage")) {
-                                        String errorMessage = obj.get("errorMessage").toString();
-                                        if (errorMessage.contains("JobValidationException")) {
-                                            errorMessage = errorMessage.substring(errorMessage.indexOf(":") + 2);
-                                        }
-                                        displayErrorMessage(errorMessage);
+                            handleVariables(obj);
+                        } catch (Throwable t) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)